### DISCUSSION TITLE
About JSON data

### DISCUSSION THREAD

**Discussion by jclinkforever** [vakata/jstree#2475] [Q&A]
Hello,
I got a question while configuring Ajax to read JSON data. 
In the demo, JSON data is expected as the following format:
`[{"id":1, "text": "Node1"}, {"id":2, "text": "Node2"},  ...]`
But in some real case, the returned JSON data(I can't change the return format) is like:
`{"lists":[{"id":1, "text": "Node1"}, {"id":2, "text": "Node2"},  ...]}`
So what should I do to read this type of JSON?
Thanks!

**Comment by vakata** [comment]
Use the ```core.data``` as a callback and make the AJAX call yourself - that way you can adapt the returned JSON to the format expected by jsTree. 

A basic demo is available here:
https://github.com/vakata/jstree/wiki#populating-the-tree-using-a-callback-function
Instead of invoking ```cb``` right away - make your AJAX call, process the data and the invoke ```cb``` with the processed data.
